test/gtest/ucs/test_bitmap.cc

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include <common/test.h>
+
+#include <ucs/datastruct/bitmap.h>
+
+class test_ucs_bitmap : public ucs::test {
+protected:
+    ucs_bitmap_t(128) bitmap;
+};
+
+void test_set_get_unset(ucs_bitmap_t(128) *bitmap, uint64_t offset)
+{
+    UCS_BITMAP_SET(bitmap, offset);
+    EXPECT_EQ(UCS_BITMAP_GET(bitmap, offset), 1);
+    EXPECT_EQ(bitmap->bits[offset >= 64], 1 << offset);
+    EXPECT_EQ(bitmap->bits[offset < 64], 0);
+
+    UCS_BITMAP_UNSET(bitmap, offset);
+    EXPECT_EQ(bitmap->bits[0], 0);
+    EXPECT_EQ(bitmap->bits[1], 0);
+    EXPECT_EQ(UCS_BITMAP_GET(bitmap, offset), 0);
+}
+
+UCS_TEST_F(test_ucs_bitmap, test_get_set_clear) {
+    const uint64_t offset = 15;
+
+    UCS_BITMAP_CLEAR(&bitmap);
+    EXPECT_EQ(bitmap.bits[0], 0);
+    EXPECT_EQ(bitmap.bits[1], 0);
+    EXPECT_EQ(UCS_BITMAP_GET(&bitmap, offset), 0);
+
+    test_set_get_unset(&bitmap, offset);
+    test_set_get_unset(&bitmap, offset + 64);
+
+    UCS_BITMAP_CLEAR(&bitmap);
+    EXPECT_EQ(UCS_BITMAP_GET(&bitmap, 117), 0);
+
+}
+
+UCS_TEST_F(test_ucs_bitmap, test_foreach) {
+    uint8_t i, word;
+    uint64_t words[2] = {};
+
+    UCS_BITMAP_CLEAR(&bitmap);
+    UCS_BITMAP_SET(&bitmap, 1);
+    UCS_BITMAP_SET(&bitmap, 25);
+    UCS_BITMAP_SET(&bitmap, 61);
+
+    UCS_BITMAP_SET(&bitmap, 64 + 15);
+    UCS_BITMAP_SET(&bitmap, 64 + 37);
+    UCS_BITMAP_SET(&bitmap, 64 + 58);
+
+    UCS_FOR_EACH_BIT(i, word, &bitmap) {
+        words[word] |= 1ull << (i % 64);
+    }
```

Comment:
why needed?

Reply:
@brminich , my goal here is to check the `UCS_BITMAP_FOR_EACH_BIT` macro. I do it by:
1. Initializing some bits in a bitmap.
2. Iterating on them using that macro. In each iteration, copy a bit to an int.
3. Compare the resulting int with the bitmap.

Assuming (1) and (3) are valid, it verifies the validity of the macro usage in (2).